livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -712,433 +702,352 @@ void Lddc::initializeDiagnostics()
-      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
-      continue;
-    }
+void Lddc::checkTemperature(diagnostic_updater::DiagnosticStatusWrapper &stat,
+                            const std::string & broadcast_code)
+{
+  int level = DiagStatus::OK;
+  std::string error_str = "OK";
+
+  const auto device_info = findDeviceInfoByBroadcastcode(broadcast_code);
 
+  if (!device_info) {
+    error_str = "disconnected";
+  } else if (device_info->state == kLidarStateInit) {
+    error_str = fmt::format("progress: %d%%", device_info->status.progress);
+  } else {
     TemperatureStatus status = static_cast<TemperatureStatus>(
       device_info->status.status_code.lidar_error_code.temp_status);
     if (status == TemperatureStatus::HighOrLow) {
       level = DiagStatus::WARN;
     } else if (status == TemperatureStatus::ExtremelyHighOrLow) {
       level = DiagStatus::ERROR;
     }
-
-    stat.add(broadcast_code, temperature_dict_.at(level));
-    whole_level = std::max(whole_level, level);
+    error_str = temperature_dict_.at(level);
   }
 
-  stat.summary(whole_level, temperature_dict_.at(whole_level));
+  stat.add("broadcast_code", broadcast_code);
+  stat.summary(level, error_str);
 }
 
-void Lddc::checkVoltage(diagnostic_updater::DiagnosticStatusWrapper & stat)
+void Lddc::checkVoltage(diagnostic_updater::DiagnosticStatusWrapper &stat,
+                        const std::string &broadcast_code)
 {
-  if (lidar_count_ == 0) {
-    stat.summary(DiagStatus::WARN, "No LiDARs Connected");
-    return;
-  }
-
-  int whole_level = DiagStatus::OK;
-
-  for (const auto & lidar : lds_->connected_lidars_) {
-    int level = DiagStatus::OK;
-
-    const auto & broadcast_code = lidar.first;
-    const auto & device_info = lidar.second;
+  int level = DiagStatus::OK;
+  std::string error_str = "OK";
 
-    if (device_info == nullptr) {
-      stat.add(broadcast_code, "disconnected");
-      continue;
-    }
-
-    if (device_info->state == kLidarStateInit) {
-      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
-      continue;
-    }
+  const auto device_info = findDeviceInfoByBroadcastcode(broadcast_code);
 
+  if (!device_info) {
+    error_str = "disconnected";
+  } else if (device_info->state == kLidarStateInit) {
+    error_str = fmt::format("progress: %d%%", device_info->status.progress);
+  } else {
     VoltageStatus status = static_cast<VoltageStatus>(
       device_info->status.status_code.lidar_error_code.volt_status);
     if (status == VoltageStatus::High) {
       level = DiagStatus::WARN;
     } else if (status == VoltageStatus::ExtremelyHigh) {
       level = DiagStatus::ERROR;
     }
-
-    stat.add(broadcast_code, motor_dict_.at(level));
-    whole_level = std::max(whole_level, level);
+    error_str = voltage_dict_.at(level);
   }
 
-  stat.summary(whole_level, voltage_dict_.at(whole_level));
+    stat.add("broadcast_code", broadcast_code);
+    stat.summary(level, error_str);
```

Comment:
Is this strange shift caused by format error?

Reply:
Oh, thaks. I modified them in https://github.com/tier4/livox_ros2_driver/pull/14/commits/5d2cfe3260ff71cfe612ea372179c7f79fb92d5a.